src/lib/clients/IOClients/IOClientFactory.ts

```diff
@@ -0,0 +1,103 @@
+import { InstanceOptions, IOClient, IOContext } from '@vtex/api'
+import { Logger } from '@vtex/api/lib/service/logger'
+import * as env from '../../../env'
+import userAgent from '../../../user-agent'
+import { SessionManager } from '../../session/SessionManager'
+
+interface IOContextOptions {
+  account?: string
+  authToken?: string
+  region?: string
+  workspace?: string
+}
+
+const noop = () => {}
+
+export class IOClientFactory {
+  public static DEFAULT_TIMEOUT = 15000
+
+  private static createDummyLogger() {
+    const { account, workspace } = SessionManager.getSingleton()
+    return ({
+      account,
+      workspace,
+      operationId: '',
+      requestId: '',
+      debug: noop,
+      info: noop,
+      warn: noop,
+      error: noop,
+      sendLog: noop,
+    } as unknown) as Logger
+  }
+
+  public static createIOContext(opts?: IOContextOptions): IOContext {
+    const session = SessionManager.getSingleton()
+    const {
+      account = session.account,
+      authToken = session.token,
+      region = env.region(),
+      workspace = session.workspace || 'master',
+    } = opts ?? {}
+
+    return {
+      account,
+      userAgent,
+      workspace,
+      authToken,
+      region,
+      production: false,
+      product: '',
+      route: {
+        id: '',
+        params: {},
+      },
+      requestId: '',
+      operationId: '',
+      platform: '',
+      logger: IOClientFactory.createDummyLogger(),
+    }
+  }
+
+  public static createClient<T extends IOClient>(
+    ClientClass: typeof IOClient,
+    customContext: Partial<IOContext> = {},
+    customOptions: Partial<InstanceOptions> = {}
+  ): T {
+    const clusterHeader = env.cluster() ? { 'x-vtex-upstream-target': env.cluster() } : null
```

Comment:
Since we're refactoring the clients. I would start caching header keys on a `constants` file.

Reply:
Perfect! Will do, and that is much better

Reply:
Done :snowboarder: